src/rshim_pcie.c

```diff
@@ -34,10 +47,18 @@
 /* The size the RShim region. */
 #define PCI_RSHIM_WINDOW_SIZE       0x100000
 
-#ifdef __linux__
-#include <dirent.h>
-#define SYS_BUS_PCI "/sys/bus/pci/devices"
-#endif
+#define SYS_CLASS_IOMMU_PATH        "/sys/class/iommu"
+#define SYS_CLASS_VFIO_PCI_PATH     "/sys/module/vfio_pci"
+#define SYS_BUS_PCI_PATH            "/sys/bus/pci/devices"
+#define SYS_VFIO_PCI_PATH           "/sys/bus/pci/drivers/vfio-pci"
+
+typedef enum {
+  RSHIM_PCIE_MMAP_DIRECT,
+  RSHIM_PCIE_MMAP_UIO,
```

Comment:
Is UIO the same as mmap_direct?

Reply:
UIO is not implemented yet and will be added as part of the interrupt support for April release.
It'll be similar to VFIO but much simpler.